src/zope/globalrequest/tests.py

```diff
@@ -1,37 +1,93 @@
-from unittest import TestSuite
-from zope.configuration import config
-from zope.configuration import xmlconfig
-from zope.testing.cleanup import cleanUp
-import doctest
-import zope.app.wsgi.testlayer
-import zope.globalrequest
-import zope.testbrowser.wsgi
+# -*- coding: utf-8 -*-
+import unittest
 
 
-def zcml(source):
-    context = config.ConfigurationMachine()
-    xmlconfig.registerCommonDirectives(context)
-    xmlconfig.string(source, context)
+class TestGlobalrequest(unittest.TestCase):
 
+    def tearDown(self):
+        # reset the threading local context
+        from threading import local
+        import zope.globalrequest.local
```

Comment:
Well, in tests I usually import test related code inside the test methods. This is probably a bit paranoid. OTOH it avoids strange and difficult to debug side effects with imports...